src/analysis/normed_space/ordered.lean

```diff
@@ -48,3 +67,146 @@ instance : normed_linear_ordered_field ℚ :=
 noncomputable
 instance : normed_linear_ordered_field ℝ :=
 ⟨dist_eq_norm, normed_field.norm_mul⟩
+
+/-!
+### Normed lattice orderd groups
+
+Motivated by the theory of Banach Lattices, this section introduces normed lattice ordered groups.
+-/
+
+local notation `|`a`|` := abs a
+
+/--
+Let `α` be a normed commutative group equipped with a partial order covariant with addition, with
+respect which `α` forms a lattice. Suppose that `α` is *solid*, that is to say, for `a` and `b` in
+`α`, with absolute values `|a|` and `|b|` respectively, `|a| ≤ |b|` implies `∥a∥ ≤ ∥b∥`. Then `α` is
+said to be a normed lattice ordered group.
+-/
+class normed_lattice_add_comm_group (α : Type*)
+  extends normed_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+(solid : ∀ a b : α, |a| ≤ |b| → ∥a∥ ≤ ∥b∥)
```

Comment:
Can you add the instance `ordered_add_comm_group` derived from this? That would also make the `...to_covariant_class` below unnecessary.

Reply:
I can add something like this, if it's what you mean?
```
instance normed_lattice_add_comm_group_to_ordered_add_comm_group  {α : Type*}
  [h : normed_lattice_add_comm_group α] : ordered_add_comm_group α := { ..h }
```
(If I omit `normed_lattice_add_comm_group_to_ordered_add_comm_group` then Lean tells me: "invalid definition, a declaration named 'ordered_add_comm_group' has already been declared")

However, if I remove `normed_lattice_add_comm_group_to_covariant_class` then I run into problems in the proof of `normed_lattice_add_comm_group_has_continuous_inf` - for example, I can't appeal to `lattice_ordered_comm_group.Birkhoff_inequalities` without proving `covariant_class α α has_add.add has_le.le`.

Perhaps I misunderstand your intention?

Reply:
I've removed the `normed_lattice_add_comm_group_to_covariant_class` instance now.